src/Pages/HomePage.php

```diff
@@ -9,7 +9,24 @@ class HomePage extends Page {
     protected $template = 'Home.phtml';
 
     protected function get() {
-        $this->data['entries'] = Entry::getEntriesForCurrentUser();
+        $filter_by = array();
+        if (isset($_REQUEST['filter_by_category_id']) && empty($_REQUEST['filter_by_category_id']) == false) {
+            $filter_by['category_id'] = intval($_REQUEST['filter_by_category_id']);
+        }
+        if (isset($_REQUEST['filter_by_from_date']) && empty($_REQUEST['filter_by_from_date']) == false) {
+            if (isset($filter_by['publish_date']) == false) {
+                $filter_by['publish_date'] = array();    
+            }
+            $filter_by['publish_date'][0] = $_REQUEST['filter_by_from_date'];
+        }
+
+        if (isset($_REQUEST['filter_by_until_date']) && empty($_REQUEST['filter_by_until_date']) == false) {
+            if (isset($filter_by['publish_date']) == false) {
+                $filter_by['publish_date'] = array();    
+            }
+            $filter_by['publish_date'][1] = $_REQUEST['filter_by_until_date'];
+        }
+        $this->data['entries'] = Entry::getEntriesForCurrentUser($filter_by);
```

Comment:
`empty` also returns false if the variable/key is not set, so we could write all

```
isset($_REQUEST['filter_by_until_date']) && empty($_REQUEST['filter_by_until_date']) == false
```

as

```
!empty($_REQUEST['filter_by_until_date'])
```

I think that would be more readable :thinking: